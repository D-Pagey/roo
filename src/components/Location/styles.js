import styled from 'styled-components';

export const Wrapper = styled.div`
    align-items: ${({ isMobile }) => (isMobile ? 'flex-start' : 'center')};
    background-color: #00ccbc;
    box-sizing: border-box;
    display: flex;
    flex-direction: ${({ isMobile }) => (isMobile ? 'column' : 'row')};
    height: ${({ isMobile }) => (isMobile ? '177px' : '156px')};
    justify-content: space-between;
    padding: 1.5rem 1rem;
`;

export const Title = styled.h4`
    font-size: 1rem;
    line-height: 1.375rem;
    margin: 0 0 2px;
    opacity: 50%;
`;

export const WhiteTitle = styled.h5`
    color: #ffffff;
    font-size: 2.125rem;
    margin: ${({ isMobile }) => (isMobile ? '0 0 1.25rem' : 0)};
`;
