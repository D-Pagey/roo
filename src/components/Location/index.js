import React from 'react';
import { string } from 'prop-types';
import { useMediaQuery } from 'react-responsive';
import { mediaQuery } from '../../tokens';
import Button from '../Button';
import * as S from './styles';

const Location = ({ location }) => {
    const isLaptopOrLarger = useMediaQuery({ query: mediaQuery.laptop });

    return (
        <S.Wrapper isMobile={!isLaptopOrLarger}>
            <S.WidthWrapper isMobile={!isLaptopOrLarger}>
                <div>
                    <S.Title>Location</S.Title>
                    <S.WhiteTitle isMobile={!isLaptopOrLarger}>{location}</S.WhiteTitle>
                </div>

                <Button>Change Location</Button>
            </S.WidthWrapper>
        </S.Wrapper>
    );
};

Location.propTypes = {
    location: string.isRequired
};

export default Location;
