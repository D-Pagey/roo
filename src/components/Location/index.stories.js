import React from 'react';
import { storiesOf } from '@storybook/react';
import Location from '.';

const props = {
    location: 'Fitzrovia'
};

storiesOf('Location', module).add('default', () => <Location {...props} />);
