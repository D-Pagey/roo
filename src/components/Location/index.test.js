import React from 'react';
import { render } from '@testing-library/react';
import Location from '.';

const props = {
    location: 'Fitzrovia'
};

describe('Location component', () => {
    it('should render', () => {
        const { container } = render(<Location {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
