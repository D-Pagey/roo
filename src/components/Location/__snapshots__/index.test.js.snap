// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location component should render 1`] = `
.c3 {
  background-color: #ffffff;
  border-radius: 3px;
  border: 0;
  color: #00a396;
  cursor: pointer;
  font-size: 1rem;
  padding: 11px 1rem;
}

.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  background-color: #00ccbc;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 177px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 1.5rem 1rem;
}

.c1 {
  font-size: 1rem;
  line-height: 1.375rem;
  margin: 0 0 2px;
  opacity: 50%;
}

.c2 {
  color: #ffffff;
  font-size: 2.125rem;
  margin: 0 0 1.25rem;
}

<div
  class="c0"
>
  <div>
    <h4
      class="c1"
    >
      Location
    </h4>
    <h5
      class="c2"
    >
      Fitzrovia
    </h5>
  </div>
  <button
    class="c3"
    type="button"
  >
    Change Location
  </button>
</div>
`;
